libtransact/src/context/mod.rs

```diff
@@ -19,14 +19,24 @@
 /// modify events, data, and state.
 pub type ContextId = [u8; 16];
 
+mod error;
 pub mod manager;
 
 use crate::receipts::Event;
 use std::mem;
 use uuid::Uuid;
 
+pub use crate::context::error::ContextManagerError;
 use crate::receipts::StateChange;
 
+/// ContextManager functionality used by the Scheduler.
+pub trait ContextLifecycle<R>: Send {
```

Comment:
We need to remove R here I think; this is not something the scheduler would/should know about.

Reply:
Yes, I would agree. I can remove it from the trait but until #49 is merged to take care of the additional R type on the ContextManager, it will be there in the implementation.

Reply:
Updated